[STN-80] Readme doc for Humsci Basic theme

nrambeck opened the pull request:
# READY FOR REVIEW

# Summary

Initial docs for the Humsci Basic theme. Tried to cover all existing functionality and relevant principles.

Also includes a11y PR checklist based on the document written in Basecamp

Commit:
docs(STN-80): add readme doc for Humsci Basic theme

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr440-oinfjdxvrhohp8enhhvb1ikgd74trmjv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743

Visual Diffs:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743/visualdiffs

silverli commented on docroot/themes/humsci/humsci_basic/readme.md:
```diff
@@ -0,0 +1,54 @@
+- `npm run build:sass` - Compile Sass for production
+- `npm run watch` - Compile a build and watch for changes
+
+## Testing
```
Let's add something about our decision on commit messages and merging as well

silverli commented on docroot/themes/humsci/humsci_basic/readme.md:
```diff
@@ -0,0 +1,54 @@
+
+## Requirements
+
+- Node 10.15+
```
and decanter 5.1.0 -> soon to be 6.0.0

silverli commented on docroot/themes/humsci/humsci_basic/readme.md:
```diff
@@ -0,0 +1,54 @@
+# Stanford Basic theme
+
+Stanford Basic is used a base for all future H&S themes. It integrates selectively with Decanter Sass, styles and 
```
We have pulled from patterns and templates in Stanford Basic (stanford_basic) as well as SU Humsci (su_humsci_theme) Themes.

silverli commented on docroot/themes/humsci/humsci_basic/readme.md:
```diff
@@ -0,0 +1,54 @@
+
+## Testing
+
+- `npm test` - Run linting and Sass tests
```
we also need to figure out how to run and make our own behat tests, and unit tests for modules

silverli commented on docroot/themes/humsci/humsci_basic/readme.md:
```diff
@@ -0,0 +1,54 @@
+
+## Sub-themes
+
+There are currently 3 sub-themes based on Humsci Basic
```
These themes reference humsci_basic css builds on a per-theme basis in their x.libraries.yml

silverli commented on docroot/themes/humsci/humsci_basic/readme.md:
```diff
@@ -0,0 +1,68 @@
+
+## Testing
+
+- `npm test` - Run linting and Sass True tests
```
we also need to figure out how to run and make our own behat tests, and unit tests for modules

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr440-oinfjdxvrhohp8enhhvb1ikgd74trmjv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743

Visual Diffs:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743/visualdiffs

silverli commented on docroot/themes/humsci/humsci_basic/readme.md:
```diff
@@ -0,0 +1,68 @@
+
+## Builds 
+
+Frontend assets are built using the Grunt task runner, but are run using npm scripts as shortcuts. Assets are compiled to a `dist/` directory with a stylesheet generated for each sub-theme.
```
we also need to figure out when we want to compile sass maps for development

silverli replied:
I can add this to a card in the current sprint

Commit:
docs(STN-80/65): adds a11y pr checklist

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr440-oinfjdxvrhohp8enhhvb1ikgd74trmjv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743

Visual Diffs:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743/visualdiffs

silverli commented on docroot/themes/humsci/humsci_basic/readme.md:
```diff
@@ -0,0 +1,68 @@
+
+- Drupal 8.7.10
+- Node 10.15+
+- Decanter 5.1.0, soon to be upgraded to 6.0.0
```
6 has been released https://github.com/SU-SWS/decanter/releases/tag/6.0.0

nrambeck replied:
Updating to say 6, since we made that change in a recent PR.

silverli commented on docs/SparkboxPRChecklist.md:
```diff
@@ -26,6 +29,37 @@
+## Accessibility
+Use this checklist when user-facing (non-admin) code is added.
+
+We will divide the work we do as part of our accessibility checklist into a 30/70 ratio of automated/manual testing, or as close as we can get.
```
I am realizing 30/70 time split doesnt make sense here 😂 but I still think that the ratio of time spent on automated vs manual testing should be less to more

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr440-oinfjdxvrhohp8enhhvb1ikgd74trmjv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743

Visual Diffs:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743/visualdiffs

bryanbraun commented on docroot/themes/humsci/humsci_basic/readme.md:
```diff
@@ -0,0 +1,68 @@
+# Humsci Basic (humsci_basic) Theme
+
+Humsci Basic (humsci_basic) is a theme that can be used as a base for all future H&S themes. We have pulled patterns and templates for Humsci Basic from Stanford Basic (stanford_basic) and Stanford University Humsci Theme (su_humsci_theme) Themes. Humsci Basic integrates selectively with Decanter Sass, styles, and patterns.
```
I'd remove the "all" here, since I could imagine us continuing to use Stanford Basic / Humsci Theme for some future site-specific themes, if it makes sense.

bryanbraun commented on docroot/themes/humsci/humsci_basic/readme.md:
```diff
@@ -0,0 +1,68 @@
+
+## Sub-themes
+
+There are currently 3 children sub-themes based on Humsci Basic. Themes themes reference built CSS files in Humsci Basic on a per-theme basis in their `{}.libraries.yml`.
```
Duplicate "Themes" in this sentence.

bryanbraun reviewed (changes requested):
Great idea. I'm glad you included this in the `humsci_basic` directory.

My biggest recommendation is to include a diagram that shows how these themes will work together, like this one:

![stanford-humsci-diagrams-D8 HumSci Multisite Themes (4)](https://user-images.githubusercontent.com/1256329/71687052-bcb80980-2d51-11ea-93c5-2d49c36bf2ec.png)

(the source file can be found in Google Drive)

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr440-oinfjdxvrhohp8enhhvb1ikgd74trmjv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743

Visual Diffs:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743/visualdiffs

nrambeck commented:
I added the diagram just under the opening paragraph. I also changed the commit and merging heading to "Contributions" and moved it closer to the bottom of the readme.

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr440-oinfjdxvrhohp8enhhvb1ikgd74trmjv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743

Visual Diffs:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743/visualdiffs

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr440-oinfjdxvrhohp8enhhvb1ikgd74trmjv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743

Visual Diffs:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743/visualdiffs

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr440-oinfjdxvrhohp8enhhvb1ikgd74trmjv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743

Visual Diffs:
- https://dashboard.tugboat.qa/5dfd24b304a8b02d40da9743/visualdiffs
